pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -164,10 +164,15 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			Fn:           flow.TaskFn(botanist.DeployETCD).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			Dependencies: flow.NewTaskIDs(deploySecrets, deployCloudProviderSecret, wailtUntilBackupEntryInGardenReconciled),
 		})
+		deployEtcdSecrets = g.Add(flow.Task{
```

Comment:
Why is this now a separate step? How is the implementation of `DeployEtcdSecrets` different from the previous implementation generating the certificates for etcd?

Reply:
Etcd statefulset pod had the form etcd-main-0 as the alternate name in the certificate. With druid, the statefulset will add a nonce to the statefulset name. So the pod name will be like etcd-main-`nonce`-0. We need to pull the name of the etcd statefulset from the status of the `etcd` resource.

Reply:
Hm, so you're saying that we first create the `Etcd` resource, then wait until the etcd-druid has created a `StatefulSet`, but this `StatefulSet` will fail because the secrets are missing yet, right? So, only after the `StatefulSet` got created we generate the secrets and then, eventually, the pods might come up? If this is true then it's quite weird. Why aren't the needed certificates generated by etcd-druid in the first place? He knows best when/what to generate.

Reply:
After a call with @georgekuruvillak we agreed on the following. The old/current name of the etcd is `etcd-main-0`, for example. In the future it might look like this: `etcd-main-jha74n-0` (the druid adds a nonce as described above).
As druid is the one deploying and managing the etcd and other components like the backup-restore sidecar, it is the only one that knows how the certificates should look like.

Hence, druid will compute a dedicated CA cert for etcd + a server and client cert. Gardener won't do this anymore. Though, Gardener needs to know the secret names of these certs as it has to inject them into the kube-apiserver because it wants to talk with etcd. Druid will expose them in the `status` of the `Etcd` resource, e.g. like this:

```yaml
...
status:
  tlsSecrets:
  - purpose: ca
    name: ca-etcd
  - purpose: client
    name: etcd-client
```